spec/controllers/shared_examples.rb

```diff
@@ -0,0 +1,92 @@
+require 'spec_helper'
+
+shared_examples :most_focus do |data_type|
```

Comment:
`shared_examples` is an Rspec thing, so I can't change the name (docs [here](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples)). The name is really set by the first argument you pass to the function (in this case `:most_focus`).

In terms of debugging, these tests behave no differently to ordinary tests, they are just a bit more general. In this case the only thing I've changed from the original tests (other than variable names to make them more general), was the `create` method calls.

Instead of `create(:moment)`, it's now `create(data_type)`. Data type is a symbol passed to the `shared_example` call in the spec (see [here](https://github.com/ifmeorg/ifme/pull/926/files#diff-f803dcdbbef9948899992eb475f5af1eR19)) and is used to generate the appropriate factories for the tests. The rest of the test remains the same.

To debug, you'd chuck a `byebug` in the spec and first check the value of `data_type` to make sure it was what you were looking for, and then proceed as you would for any other spec.

Reply:
Oh gotcha, that's fine by me then. I am a little concerned that this is too metaprogrammy, but after some research, I think it should be fine.

 Thanks for explaining, go ahead and merge whenever you're ready.